cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGenerator.scala

```diff
@@ -0,0 +1,654 @@
+      temporalSchemaMap.put(name, schema.asInstanceOf[Schema])
+    }
+
+    schemaMap.filter{ case(name, schema) =>
+      schema.isInstanceOf[LookupSchema]
+    }.foreach{ case(name, schema) =>
+      lookupSchemaMap.put(name, schema.asInstanceOf[LookupSchema])
+    }
+
+    (temporalSchemaMap.toMap, lookupSchemaMap.toMap)
+  }
+
+  def parseCreate(create: CreateView, schemaMap: Map[String, AbstractSchema]): String = {
+    val (temporalSchemaMap, lookupSchemaMap) = splitSchemaMap(schemaMap)
+    val sourceSchema = temporalSchemaMap(create.query.dataset)
+    val resultSchema = calcResultSchema(create.query, sourceSchema)
+    val ddl: String = genDDL(create.dataset, sourceSchema)
+    val insert =
+      s"""
+     |replace into $quote${create.dataset}$quote ()
+     |(
+     |${parseQuery(create.query, schemaMap)}
+     |)
+      """.stripMargin.replace("T", " ").replace("Z", "")
+    ddl + insert
+  }
+
+  protected def genDDL(name: String, schema: Schema): String = {
+    def mkNestDDL(names: List[String], typeStr: String): String = {
+      names match {
+        case List(e) => s"  $e $typeStr"
+      }
+    }
+
+    val fields = schema.fieldMap.values.filter(f => f.dataType != DataType.Hierarchy && f != AllField).map {
+      f => mkNestDDL(f.name.split("\\.").toList, fieldType2ADMType(f) + (if (f.isOptional) " default null" else " not null"))
+    }
+    s"""
+       |create table if not exists $quote${name}$quote (
+       |${fields.mkString(",\n")}, primary key (${schema.primaryKey.map(key => key.name).mkString(",")})
+       |);
+      """.stripMargin
+  }
+
+  protected def fieldType2ADMType(field: Field): String = {
+    field.dataType match {
+      case DataType.Number => "double"
+      case DataType.Time => "datetime"
+      case DataType.Point => "point"
+      case DataType.Boolean => "tinyint"
+      case DataType.String => "varchar(255)"
+      case DataType.Text => "text"
+      case DataType.Json => "json"
+      case DataType.Bag => s"{{${fieldType2ADMType(Field("", field.asInstanceOf[BagField].innerType))}}}"
+      case DataType.Hierarchy => ???
+      case DataType.Record => ???
+    }
+  }
+
+  def parseAppend(append: AppendView, schemaMap: Map[String, AbstractSchema]): String = {
+    s"""
+       |replace into $quote${append.dataset}$quote (
+       |${parseQuery(append.query, schemaMap)}
+       |)""".stripMargin.replace("T", " ").replace("Z", "")
+  }
+
+  def parseUpsert(q: UpsertRecord): String = {  //TODO: generalization
+    val x = q.records.value
+    var queryResult = ArrayBuffer.empty[String]
+    x.foreach {
+      record =>
+        val name: String = (record \ "name").as[JsString].value
+        val schema: JsValue = (record \ "schema").as[JsValue]
+        val dataInterval: JsValue = (record \ "dataInterval").as[JsValue]
+        val stats: JsValue = (record \ "stats").as[JsValue]
+        queryResult += (s"('${name}','${schema}','${dataInterval}','${stats}')")
+    }
```

Comment:
Since parseUpsert() is only called when inserting schemas into berry.meta, and the fields are fixed.
Do I have to generalize this function?